huntsman/observatory.py

```diff
@@ -496,14 +483,15 @@ def _take_flat_fields(self, exptimes, observation, fits_headers=None, dark=False
         for cam_name, exptime in exptimes.items():
             cam = self.cameras[cam_name]
 
-            # Define filename for flat fields
-            filename = (f'{image_dir}/flats/{cam.uid}/{observation.seq_time}'
-                        f'/{imtype}_{observation.current_exp_num:02d}.{cam.file_extension}')
+            # Create filename
+            filename = os.path.join(observation.directory, f'{imtype}_'
+                                    f'{observation.current_exp_num:02d}.{cam.file_extension}')
```

Comment:
Breaking up the filename in the middle of the word is odd.

```suggestion
            filename = f'{imtype}_{observation.current_exp_num:02d}.{cam.file_extension}'
            filename = os.path.join(observation.directory, filename)
```

This would make even more sense with:

```py
filename = 'blah'
path = os.path.join(obs.directory, filename)
```

But we haven't been very about using proper terminology w.r.t. filenames vs pathnames. Here is is a little odd because `take_observation` can do handle either.